build/AzurePipelinesTemplates/WindowsAppSDK-PublishInsightsHeader-Steps.yml

```diff
@@ -0,0 +1,10 @@
+steps:
+# START of copying files into their final package location.
+- task: CopyFiles@2
+  displayName: 'copy WindowsAppSDKInsights.h into output folder'
+  inputs:
+    SourceFolder: '$(Build.SourcesDirectory)\dev\WindowsAppSDK_Insights'
+    Contents: |
+      WindowsAppSDKInsights.h
+    TargetFolder: '$(Build.ArtifactStagingDirectory)\fullnuget\include'
+    flattenFolders: true
```

Comment:
I don't think it's justified to have a separate job for copy one file. This needs to be part of previous "build" job.

Reply:
I disagree. Each module should have its own build/publish yml. The uber yml should only invoke and aggregate the results of child yml.

Reply:
restore of telemetry nuget package is part of reunion build job, but prep the telemetry header file for publish is a separate job. Also it's very inefficient to do the copy in its own job. @kythant

Reply:
@ZenBird Insights isn't a module, it's an API in WinAppSDK. If it's becoming 1 .h file then...
1. it should live at `/dev/Insights` (just contains the 1 header file)
2. Update `dev/WindowsAppSDK_DLL/WindowsAppSDK_DLL.vcxproj` to copy ('publish') the header like others
3. CopyFilesToStagingDir.ps1 adds a line to copy the header to the Transport package (line it does MsixDynamicDependency.h and other headers)

WindowsAppSDK_DLL.vcxproj update (end of the file)

```
...
  <Target Name="CopyFiles" AfterTargets="AfterBuild">
    <Copy SourceFiles="@(PublicHeaders)" DestinationFolder="$(OutDir)" />
    <Copy SourceFiles="$(MSBuildThisFileDirectory)..\Insights\WindowsAppSDKInsights.h" DestinationFolder="$(OutDir)" />
  </Target>
</Project>
```

This is consistent with how other WinAppSDK APIs' headers are handled.

It also makes Insight's header/API available to others in the project e.g. WindowsAppSDK_BootstrapDLL project will want to reference this soon. Cross-project should reference 'published' artifacts and not reach into source dirs (take a look at dev/DynamicDependency - there's 3 headers meant for use by others, everything else is implementation detail. But if folks used them from here they're point to the directory and have no clear distinction which headers are implementation vs interface).

Reply:
@huichen123 you do see the irony, given MRT is its own separate beast bolted onto Foundation (`/dev/mrt`, `build-mrt.yml`, `publish-mrt.yml`)? ;-)

Is that a temporary matter you expect to be (eventually) merge into WindowsAppSDK.sln, CopyFilesToStagingDir.ps1, etc?

Reply:
merging into WindowsAppSDK.sln is low priority. I'd only do that when WindowsAppSDK.sln has no build error/warning.